### PR TITLE
feat: add oauth2 client_type field

### PR DESCRIPTION
**Pull request by onematchfox**
Adds support for the `client_type` field introduced in https://github.com/Kong/kong/pull/5268.

### CODE REVIEW

File: kong/oauth2_auth_service_test.go [hunk 1]
```diff
@@ -88,6 +88,43 @@ func TestOauth2CredentialCreateWithID(T *testing.T) {
 	assert.NoError(client.Consumers.Delete(defaultCtx, consumer.ID))
 }
 
+func TestOauth2CredentialCreatePublicClientType(T *testing.T) {
+	assert := assert.New(T)
+	require := require.New(T)
+
+	client, err := NewTestClient(nil, nil)
+	assert.NoError(err)
+	assert.NotNil(client)
```
Comment (pmalek) [inline kong/oauth2_auth_service_test.go:97]:
```suggestion
	require.NoError(err)
	require.NotNil(client)
```

File: kong/oauth2_auth_service_test.go [hunk 2]
```diff
@@ -88,6 +88,43 @@ func TestOauth2CredentialCreateWithID(T *testing.T) {
...
+	}
+
+	consumer, err = client.Consumers.Create(defaultCtx, consumer)
+	assert.NoError(err)
+	require.NotNil(consumer)
+
+	createdOauth2Credential, err := client.Oauth2Credentials.Create(
+		defaultCtx, consumer.ID, oauth2Cred)
+	assert.NoError(err)
+	assert.NotNil(createdOauth2Credential)
```
Comment (pmalek) [inline kong/oauth2_auth_service_test.go:120]:
```suggestion
	require.NoError(err)
	require.NotNil(createdOauth2Credential)
```

File: kong/oauth2_auth_service_test.go [hunk 3]
```diff
@@ -88,6 +88,43 @@ func TestOauth2CredentialCreateWithID(T *testing.T) {
...
+		RedirectURIs: StringSlice("http://foo.com", "http://bar.com"),
+	}
+
+	// consumer for the oauth2 cred
+	consumer := &Consumer{
+		Username: String("foo"),
+	}
+
+	consumer, err = client.Consumers.Create(defaultCtx, consumer)
+	assert.NoError(err)
```
Comment (pmalek) [inline kong/oauth2_auth_service_test.go:114]:
```suggestion
	require.NoError(err)
```

**Review by pmalek — COMMENTED** [review]
Thanks! Some minor nits in tests (no point in continuing when we e.g. can't create a client) but overall this looks good 👍

**Comment by onematchfox** [comment]
> Thanks! Some minor nits in tests (no point in continuing when we e.g. can't create a client) but overall this looks good +1

Thanks for the quick review! Changes applied. For reference, I had copied the usage of `assert` from the [existing tests](https://github.com/Kong/go-kong/blob/e3c8f5b2e8dc917e855c5df39e1cdc958dcdc61d/kong/oauth2_auth_service_test.go#L53-L89) so you may want to consider refactoring these.

**Comment by pmalek** [comment]
> For reference, I had copied the usage of `assert` from the [existing tests](https://github.com/Kong/go-kong/blob/e3c8f5b2e8dc917e855c5df39e1cdc958dcdc61d/kong/oauth2_auth_service_test.go#L53-L89) so you may want to consider refactoring these.

Yup, we're aware of that. One thing at a time :) Thanks!

**Review by pmalek — APPROVED** [review]

**Pull request merged**